### ISSUE TITLE
direct use of WiFiClient ?

### ISSUE THREAD

**Issue by real-bombinho** [hsaturn/TinyMqtt#76]
Hello, I just forked your nice bit of work (thank you) and modified it to accommodate for W5500 LAN adaptors. Pretty much anything supported by the Ethernet library should work too. But I have not yet included other controllers than the ESP8266.
Thanks to you the adaption was fairly straight forward, mostly just needing a "using TcpClient = EthernetClient;" + server and some initialisation.

I noticed 2 instances of direct use of WiFiClient instead of TcpClient in TinyMqtt.cpp (lines 66 & 214 in the current main). Is there a particular reason? I changed them in my fork and it worked fine so far.

Suggestion: 
change 
  tcp_client = new WiFiClient(*new_client);
to
  tcp_client = new TcpClient(*new_client);

and change
  WiFiClient client = server->accept();
to
  TcpClient client = server->accept();

(Another problem I had, I could neither get the master or the later releases to work. They compiled fine but none of my clients connected to the broker. So I went back to the working release from the Arduino library manager, which I know to be working fine. I must admit, I did not spend a great deal of time yet on fault finding.) <= side note only

My main intention was to get it working on the ESP + W5500 first as I intended to port it to the RP2040 W5500-EVB-Pico at a later point. 

Edit: Which I have done now.

**Comment by hsaturn** [comment]
Hello, there is absolutely no reason.
Thanks for reporting this.

Best regards.

**Comment by hsaturn** [comment]
Closed by https://github.com/hsaturn/TinyMqtt/commit/25602ec2f53d84e677b2f36a464849979ae69286

**Comment by hsaturn** [comment]
Reopened because of this.

(Another problem I had, I could neither get the master or the later releases to work. They compiled fine but none of my clients connected to the broker. So I went back to the working release from the Arduino library manager, which I know to be working fine. I must admit, I did not spend a great deal of time yet on fault finding.) <= side note only

Could you provide more information please ?

**Comment by real-bombinho** [comment]
I had used the Library manager for my first attempt and compilation worked fine and Clients connected straight away.
Then I replaced the libraries with the ones from the current main. Everything compiled fine but Clients did not connect to the broker. As I had seen main branches fail before I picked the latest release, 1.0.0. Same problem. Going back to the libraries from the Arduino library manager brought the clients straight back on.

In the mean time I noticed #71 , seemingly forcing [richievos](https://github.com/richievos) onto an older version too. I have not retried yet, but it would be worth trying to see if removing the -5000 would bring it live again.
